addons/pos_adyen/static/src/js/payment_adyen.js

```diff
@@ -10,8 +10,25 @@ var _t = core._t;
 
 var PaymentAdyen = PaymentInterface.extend({
     send_payment_request: function (cid) {
+        var self = this;
         this._super.apply(this, arguments);
         this._reset_state();
+        rpc.query({
+            model: 'pos.payment.method',
+            method: 'get_diagnosis_status',
+            args: [[this.payment_method.id]],
+        }, {
+            // When a payment terminal is disconnected it takes Adyen
+            // a while to return an error (~6s). So wait 10 seconds
+            // before concluding Odoo is unreachable.
+            timeout: 10000,
+            shadow: true,
+        }).then(function (diagnosis_response) {
+            if (diagnosis_response.SaleToPOIResponse.DiagnosisResponse.Response.Result === 'Failure') {
+                self._show_error(_t('The connection to your payment terminal failed. Please check if it is still connected to the internet.'));
+            }
+        });
+
         return this._adyen_pay(cid);
```

Comment:
Don't we need to know the diagnosis before sending the actual payment request? If so, we should put `this._adyen_pay(cid)` inside `.then`.